src/time/Waveform.cpp

```diff
@@ -58,10 +59,12 @@ Eigen::VectorXd Waveform::sample(double normalizedDt)
 void Waveform::moveToNextWindow()
 {
   PRECICE_ASSERT(_storageIsInitialized);
-  auto initialGuess = extrapolate();
-  utils::shiftSetFirst(this->_timeWindowsStorage, initialGuess); // archive old samples and store initial guess
-  if (_numberOfStoredSamples < sizeOfSampleStorage()) {          // together with the initial guess the number of stored samples increases
-    _numberOfStoredSamples++;
+  if (_extrapolationOrder != Time::UNDEFINED_EXTRAPOLATION_ORDER) { // @todo: we should define the extrapolation data for all data (if the waveform is owned by the mesh::Data), another argument for configuring extrapolation order there.
```

Comment:
Currently, `_extrapolationOrder` is a property of a coupling scheme. Not all data is extrapolated (not data sent from first to second in a serial coupling scheme). And different coupling schemes can have different orders. Do you have the latter on the radar?